### DISCUSSION TITLE
Failed to start smartapplianceenabler.service

### DISCUSSION THREAD

**Discussion by ThomasKupfer75** [camueller/SmartApplianceEnabler#612] [Installation und Betrieb]
Hallo,
seit einigen Tagen funktioniert der SmartApplianceEnabler bei mir nicht mehr, davor hat alles prima funktioniert.
Ich habe dann die Installation erneut durchgeführt, leider ohne Verbesserung. ->**Failed to start smartapplianceenabler.service**
Ich kenne mich nicht so sehr mit Linux aus, aber ich habe versucht den SmartApplianceEnabler zu starten -> sudo systemctl start smartapplianceenabler -> **Failed to start smartapplianceenabler.service: Unit mosquitto.service not found**.
Kann mir jemand weiterhelfen das Problem zu lösen?

**Comment by camueller** [comment]
Hast Du den MQTT-Server, den der SAE benötigt, auf einem anderen System laufen? Falls ja, kann Du die Zeile
```
Requires=mosquitto.service
```
in der Datei `/lib/systemd/system/smartapplianceenabler.service` löschen und danach `sudo systemctl daemon-reload`ausführen.

Falls Du keinen MQTT-Server woanders laufen hast, musst Du ihn installieren (passiert in der Installation eigentlich automatisch): 
https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/InstallationManual_DE.md#mqtt-broker

**Comment by ThomasKupfer75** [comment]
Hallo Axel,
vielen Dank für deine rasche Antwort.
Ich habe den MQTT-Server auf keinem anderen System laufen.
Der Versuch den MQTT-Server manuell zu installieren hat leider nicht funktioniert, habe es auch mit --fix-broken install versucht, siehe Sceenshot im Anhang.
![20240612_155936](https://github.com/camueller/SmartApplianceEnabler/assets/106485401/90a7cbf5-acde-48ba-a109-8ad242f752ab)

**Reply by camueller** [reply]
Hast Du vor der Installation ein `sudo apt update`gemacht? Falls nicht, solltest Du es tun und die Installation nochmal starten. Anonsten ist mir rätselhaft, wie Du zu den kaputten Dependecies kommst.

**Comment by ThomasKupfer75** [comment]
Hallo Axel,
das Update hat leider auch nichts genutzt.
Hier mal noch ein paar Dinge die mir während der Installation aufgefallen sind.
- Ich verwende einen Raspberry 3 A+
- Der erste Schritt (booten von SD-Karte) dauert bei mir wesentlich länger als 1 Minute, es sind bei mir fast 15 Minute bis die grüne LED min. 10 Sekunden nicht mehr leuchtet. 
- Die Konfiguration auf dem USB-Stick (WLAN, Tastatursprache) scheint nicht übernommen zu werden. Auch scheint der SAE beim hochfahren nicht gestartet zu werden. Wenn ich direkt im Raspberry Pi Imager die Daten eingebe werden sie übernommen, d.h. WLAN... funktionieren.
- Die erste Installation vor ein paar Monaten hat auf anhieb funktioniert und hat bis vor einigen Tagen perfekt funktioniert. 

Vielleicht fällt dir dazu noch etwas ein?
Danke & Gruß

**Comment by camueller** [comment]
Dein Eingangspost war ja, dass Mosquitto "plötzlich" nicht mehr da war. Ich habe keine Idee, wie Du das geschafft hast, aber eigenlich hättest Du ihn nur wieder  installieren/starten müssen.

Dass sich der SAE nicht ohne Mosquitto starten lässt, sollte Dich nicht verwundern. Sonst könnte man ihn auch gleich weglassen.

Ich denke, dass Du die [manuelle Installation](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/InstallationManual_DE.md) probieren solltest. Schritt für Schritt jeden Befehl ausführen (copy & paste) in einer SSH-Shell. Wenn Du das nicht willst oder kannst, kann ich Dir alles installieren im Rahmen des [kostenpflichtigen Supports](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Support_DE.md). Dazu müsstest Du mir Deinen Pi schicken.

**Reply by msvintaudio** [reply]
Habe das gleiche Problem. Mosqitto kann nicht gestartet werden. 
Ansonsten auch Raspi 3+, script gesteuerte Installation.
Ebenso klappt die automatisierte Übernahme der USB Daten nicht.
Auch bei mir war plötzlich eine lange stabil laufende Version nicht mehr startfähig....
Auffällige Parallelen....

**Reply by msvintaudio** [reply]
....und die abschliessende webmin Installation scheint auch zu hängen

Hier in der Konsole:
+ echo ********** Installing Webmin ...
+ wget https://sourceforge.net/projects/webadmin/files/webmin/2.105/webmin_2.105_all.deb/download -O /tmp/webmin_2.105_all.deb
+ dpkg -i /tmp/webmin_2.105_all.deb
dpkg: dependency problems prevent configuration of webmin:
 webmin depends on libnet-ssleay-perl; however:
  Package libnet-ssleay-perl is not installed.
 webmin depends on libauthen-pam-perl; however:
  Package libauthen-pam-perl is not installed.
 webmin depends on libio-pty-perl; however:
  Package libio-pty-perl is not installed.

dpkg: error processing package webmin (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 webmin
+ sudo systemctl start webmin
Failed to start webmin.service: Unit webmin.service not found.
+ sleep 15
+ sudo systemctl status webmin
Unit webmin.service could not be found.
+ [ true = true ]
+ echo ********** Installing mbusd ...
+ cd /tmp
+ git clone https://github.com/camueller/mbusd.git
/usr/local/bin/install2.sh: 124: git: not found
+ cd mbusd
/usr/local/bin/install2.sh: 125: cd: can't cd to mbusd
+ mkdir build
+ cd build
+ cmake -DCMAKE_INSTALL_PREFIX=/usr ..
/usr/local/bin/install2.sh: 128: cmake: not found
+ make
make: *** No targets specified and no makefile found.  Stop.
+ make install
make: *** No rule to make target 'install'.  Stop.
+ systemctl daemon-reload
+ cp /etc/mbusd/mbusd.conf.example /etc/mbusd/mbusd-ttyUSB0.conf
cp: cannot stat '/etc/mbusd/mbusd.conf.example': No such file or directory
+ echo ********** Clean up installation files ...
+ rm -v /etc/rc.local
+ mv -v /etc/rc.local.bak /etc/rc.local
mv: cannot stat '/etc/rc.local.bak': No such file or directory
+ rm -v /usr/local/etc/install.config
+ echo ********** /usr/local/bin/install2.sh finished
/usr/local/bin/install2.sh: 143: cannot create /sys/class/leds/led1/brightness: Directory nonexistent
+ sleep 3600

**Reply by msvintaudio** [reply]
...habe nun die händische Installation versucht.
Bereits bei sudo apt upgrade wird auf fehlerhafte Abhängigkeiten hingewiesen, die sich aber mit dem dort angegebenen Befehl noch beheben lassen.
Aber (und das wird auch das script betreffen): es gibt eine Fehlermeldung bei der Java-installation, die ich derzeit nicht beheben kann:

The following packages have unmet dependencies:
 ca-certificates-java : Breaks: openjdk-11-jre-headless (< 11.0.19+7~1~) but 11.0.17+8-2 is to be installed
E: Unable to correct problems, you have held broken packages.

Da weiss ich nicht weiter, und ohne ein funktionierendes Java wirds nicht laufen. Schade.

**Reply by msvintaudio** [reply]
Jetzt läuft es ;-)
Zuerst:
"apt dist-upgrade" (Aktualisierung der distro info's).
Dann darf nicht der Java11 Befehl aus der manuellen Installationsanleitung  benutzt werden sondern:
"sudo apt install default-jdk"
Danach die Version überprüfen: "sudo java --version"
und der .war Befehl für den download der 2.3.1 Version lautet:
"sudo wget https://github.com/camueller/SmartApplianceEnabler/releases/download/2.3.1/SmartApplianceEnabler-2.3.1.war -P /opt/sae"
Die Kosnolenbefehle habe ich in Gänsefüsschen gesetzt, NICHT diese übernehmen.

**Reply by msvintaudio** [reply]
...nun habe ich aber doch eine Frage, da mqtt nicht korrekt ausgelesen wird.
im mqtt explorer zeigt sich folgende Info:
![Bildschirmfoto 2024-06-14 um 09 04 17](https://github.com/camueller/SmartApplianceEnabler/assets/126255744/f42aea1e-5467-49de-8966-c73509a60be8)
Eingestellt im SAE habe ich:
![Bildschirmfoto 2024-06-14 um 09 06 03](https://github.com/camueller/SmartApplianceEnabler/assets/126255744/e8d64cad-83c2-41a0-8bf6-3f7328756dbd)
Übermittel als Verbraucher wird aber die Obergrenze des Verbrauchers in Watt:

**Comment by msvintaudio** [comment]
![Bildschirmfoto 2024-06-14 um 09 11 52](https://github.com/camueller/SmartApplianceEnabler/assets/126255744/d96c513b-6f19-42f7-9282-7766dba5253f)

**Comment by ThomasKupfer75** [comment]
Super, vielen Dank für diese Infos "msvintaudio", nun funktionierts bei mir auch wieder :-)

**Comment by msvintaudio** [comment]
Gern geschehen  - das Projekt SAE ist es auf jedem Fall wert!!

**Comment by sgr002** [comment]
Danke für den Tipp mit Java, da hat mir bei einer Neuinstalltion auch gefehlt.
